pkg/watch/list.go

```diff
@@ -0,0 +1,101 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watch
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/syncutil"
+
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func randSuffix(n int) string {
+	var out strings.Builder
+
+	for i := 0; i < n; i++ {
+		j := rand.Intn(26)
+		out.WriteRune(rune('a' + j))
+	}
+	return out.String()
+}
+
+// List lists resources of a particular kind using a watch Manager (informers are shared with dynamic watchers).
+func List(ctx context.Context, wm *Manager, gvk schema.GroupVersionKind, cbForEach func(runtime.Object)) error {
+	initialPopulation := make(chan event.GenericEvent, 1024)
+	name := "list-" + gvk.Kind + randSuffix(8)
```

Comment:
I agree this adds extra complexity without much benefit. If we want to try to do something like this, we should put the registrar on the constraint template tracker and config tracker, which would guarantee that its lifespan interacts with the dynamic sync registrars. It would also avoid creating random names.

Finally, we should be fine not to share the same cache as watchmanager, as this appears to be adding complexity to an already complex bit of code. I think I'd be more comfortable if we didn't touch watch manager TBH.